ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -869,8 +921,429 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
         // act_block_w_div == 1 is currently the default value.
         conv_config.act_block_w_div = tt::div_up(in_channels, width_sharded_num_cores * constants::TILE_WIDTH);
     }
+    return conv_config;
+}
+
+std::tuple<OptimizedConvParallelizationConfig, OptimizedConvBlockConfig, MemoryConfig> get_conv_configs(
+    const Conv2dConfig& conv_config,
+    const DeviceComputeKernelConfig& compute_config,
+    const ParallelConfig& input_parallel_config,
+    const ParallelConfig& output_parallel_config,
+    TensorMemoryLayout shard_layout,
+    uint32_t in_channels,
+    uint32_t out_channels,
+    uint32_t batch_size,
+    uint32_t input_height,
+    uint32_t input_width,
+    uint32_t output_height,
+    uint32_t output_width,
+    std::array<uint32_t, 2> kernel_size,
+    uint32_t max_num_cores_c) {
+    auto elem_size = conv_config.weights_dtype == DataType::BFLOAT8_B ? 1 : 2;
+    bool is_non_tile_mul_width =
+        (conv_config.shard_layout == TensorMemoryLayout::BLOCK_SHARDED) && conv_config.act_block_h_override == 0 &&
+        (conv_config.weights_dtype == DataType::BFLOAT8_B || conv_config.weights_dtype == DataType::BFLOAT16) &&
+        conv_config.output_layout == Layout::ROW_MAJOR && ((elem_size * in_channels) % (16 * max_num_cores_c)) == 0;
+    bool use_non_tile_height = (shard_layout == TensorMemoryLayout::HEIGHT_SHARDED) && out_channels <= 256 &&
```

Comment:
use_non_tile_height should logic for this be factored into a function and used here in main code path and in prepare weights?